ligand/views.py

```diff
@@ -44,26 +112,757 @@ class LigandNameSelection(AbsTargetSelection):
+                ligand_data_potency = result['potency_data']
+            else:
+                ligand_data_affinity,ligand_data_potency = LigandListDetails(mode, ps,ligand_search=True)
+            context = {}
+            context['ligand_query'] = True
+            context['potency_data'] = ligand_data_potency
+            context['affinity_data'] = ligand_data_affinity
+            if len(ligand_data_potency) + len(ligand_data_affinity) > MAX_RECORDS:
+                context = "redirect"
+                session_key = 'ligand_bulk_search_error_msg'
+                if search_type == 'id':
+                    session_key = 'ligand_bulk_search_by_id_error_msg'
+                elif search_type == 'inchikey':
+                    session_key = 'ligand_bulk_search_error_msg'
+                elif search_type == 'names':
+                    session_key = 'ligand_bulk_search_by_names_error_msg'
+                elif search_type == 'name':
+                    session_key = 'ligand_bulk_search_by_name_error_msg'              
+                self.request.session[session_key] = 'Too many records (max. %d). Try again with more specific search criteria.' % (MAX_RECORDS)
+                self.request.session.modified = True
+                return context
+            cache.set(cache_key, context, 60*60*24*7)
+        context['mode'] = mode
+        return context
+    
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        if context == "redirect":
+            return redirect("ligand_selection")
+        elif 'redirect_to' in context:
+            return redirect(context['redirect_to'])
+        return self.render_to_response(context)
+    
+def getLigandStructuralSearchParameters(request):
+    param_dict = {}
+    default_smiles = ''
+    param_dict['search_type'] = request.session.get('ligand_structural_search_search_type', 'similarity')
+    # if param_dict['search_type'] == 'similarity':
+    #     default_smiles = "O=C1OC2C3([C@H]1OCc1ccccc1)C([C@@H](C1C43[C@@](O2)(C(=O)O1)[C@@]1([C@H](C4O)OC(=O)C1C)O)F)C(C)(C)C"
+    # else:
+    #     default_smiles = 'cccccc'
+    param_dict['input_type'] = request.session.get('ligand_structural_search_input_type', 'smiles')
+    param_dict['smiles'] = request.session.get('ligand_structural_search_smiles', default_smiles)
+    param_dict['similarity_threshold'] = float(request.session.get('ligand_structural_search_similarity_threshold', 0.5))
+    # param_dict['search_limit'] = request.session.get('ligand_structural_search_limit ', 100)
+
+    param_dict['stereochemistry'] = request.session.get('ligand_structural_search_stereochemistry',False)
+    return param_dict
+
+def setLigandStructuralSearchParameters(request, search_params_data):
+    for key,value in search_params_data.items():
+        request.session["ligand_structural_search_"+key] = value 
+    request.session.modified = True
+
+""" class LigandStructuralSelection(AbsTargetSelection):
```

Comment:
please remove class if not needed

Reply:
solved in [60f6a84](https://github.com/protwis/protwis/pull/550/commits/60f6a84db9a0834025d1ce5ba90db9e26d53c811)